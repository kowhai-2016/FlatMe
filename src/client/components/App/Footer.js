import React from 'react'
import { Link } from 'react-router'

const Footer = props => {
  return (
    <div>
      <nav className='Footer navbar navbar-inverse navbar-bottom'>
        <div className='container-fluid signature' >
          <div><h6><Link to='/about'>Kowhai</Link>&emsp;&emsp;&emsp;&emsp;&emsp;2017</h6></div>
          <a href='https://github.com/kowhai-2016/flatme'>
            <img className='git-logo center-block' src='/images/GitHub-Mark-light-32px.png' />
          </a>
        </div>
      </nav>
    </div>
  )
}

export default Footer
