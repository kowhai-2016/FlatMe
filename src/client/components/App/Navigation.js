import React, { PropTypes } from 'react'
import { Navbar, Nav, NavItem, Modal, Button } from 'react-bootstrap'
import { actions, Control, Errors, Form } from 'react-redux-form'
import { LinkContainer } from 'react-router-bootstrap'
import { browserHistory } from 'react-router'

const Navigation = React.createClass({
  getInitialState () {
    return {show: false}
  },

  open () {
    this.setState({show: true})
  },

  close () {
    this.setState({show: false})
  },

  handleLogout () {
    this.props.logout()
    browserHistory.push('/')
  },

  onSubmit (values) {
    const { email, password } = values
    this.props.login(email, password)
      .then(action => {
        if (action.type === 'LOGIN_SUCCESS') {
          this.close()
        } else {
          this.props.dispatch(actions.setErrors('forms.loginCredentials', 'The email address or password that you\'ve entered is incorrect'))
        }
      })
  },

  toUserPage (user) {
    if (user) {
      browserHistory.push(`/user/${user.id}`)
    }
  },

  render () {
    const user = this.props.account.user
    const loggedInButtons = (
      <Nav pullRight bsStyle='pills'>
        <NavItem eventKey={1} onClick={() => this.toUserPage(user)}>{user ? user.firstName : null}</NavItem>
        <NavItem eventKey={2} onClick={this.handleLogout}>Log out</NavItem>
      </Nav>
    )
    const loggedOutButtons = (
      <Nav pullRight bsStyle='pills'>
        <LinkContainer to='/join'><NavItem eventKey={1}>Sign Up</NavItem></LinkContainer>
        <NavItem eventKey={2} onClick={this.open}>Log in</NavItem>
      </Nav>
    )
    const loginModal = (
      <Modal show={this.state.show} onHide={this.close}>
        <Form model='forms.loginCredentials' onSubmit={this.onSubmit}>
          <Modal.Header closeButton>
            <Modal.Title>Log in</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <div className='loginForm'>
              <div>
                <label>Email:</label>
                <Control.text className='loginEmailField' model='.email' />
              </div>
              <div>
                <label>Password:</label>
                <Control model='.password' type='password' />
              </div>
            </div>
            <Errors model='forms.loginCredentials' />
          </Modal.Body>
          <Modal.Footer>
            <Button type='submit'>Send</Button>
          </Modal.Footer>
        </Form>
      </Modal>
    )
    return (
      <div>
        <Navbar inverse collapseOnSelect>
          <Navbar.Header>
            <Navbar.Brand>
              <a href='/'><img src='/images/web.svg' width='30px' /></a>
            </Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
          <Navbar.Collapse>
            {user ? loggedInButtons : loggedOutButtons}
          </Navbar.Collapse>
        </Navbar>
        {loginModal}
      </div>
    )
  }
})

Navigation.propTypes = {
  dispatch: PropTypes.func.isRequired,
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired
}

export default Navigation
