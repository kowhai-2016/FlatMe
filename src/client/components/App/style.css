.Footer .git-logo{
  margin-top: 8px;
  width:20px;
}

.Footer .signature h6{
  margin-top: 17px;
  text-align: center;
  color: lightgrey;
  font-size: 10px;
  margin-bottom: -25px;
  margin-left: -15px;
}

.Navigation .navbar {
  margin-bottom: 0px;
  font-family: 'Cabin Sketch', cursive;
  font-size: 1.3em;
}

.Navigation .navbar-brand img {
  display: inline-block;
}

.Navigation .navbar-brand span {
  display: inline-block;
  padding-left: 15px;
  font-size: 1.3em;
}

.Navigation .navbar-brand {
  width: 200px;
}


/* Login Modal
–––––––––––––––––––––––––––––––––––––––––––––––––– */
.App .loginForm{
  display: flex;
  justify-content: space-between;
}

.App .loginForm label {
  padding-right: 10px;
}
