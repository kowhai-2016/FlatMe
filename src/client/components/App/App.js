import React from 'react'

import Navigation from '../../containers/Navigation'

const App = props => {
  return (
    <div>
      <Navigation />
      {props.children}
    </div>
  )
}

export default App
