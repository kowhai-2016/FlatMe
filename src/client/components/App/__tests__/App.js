/* eslint-env jest */

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'

import App from '..'

test('App renders correctly', () => {
  const wrapper = shallow(<App />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
