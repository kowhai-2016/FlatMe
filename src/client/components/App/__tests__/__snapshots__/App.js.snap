exports[`test App renders correctly 1`] = `
<div>
  <Navigation />
</div>
`;
